README.md

```diff
@@ -141,3 +141,23 @@ number.
 
 - run: echo "Deno version is ${{ steps.deno.outputs.deno-version }}"
 ```
+
+### Caching dependencies downloaded by Deno automatically
+
+Dependencies installed by Deno can be cached automatically, which is similar to
+the [`cache` option in `setup-node`](https://github.com/actions/setup-node).
+
+To enable the cache, use `cache: true`. It's recommended to also add the
+`cache-hash` property, to scope caches based on lockfile changes.
+
+```yaml
+- uses: denoland/setup-deno@v2
+  with:
+    cache: true
+    cache-hash: ${{ hashFiles('**/deno.lock') }}
+```
```

Comment:
I wonder if maybe this should be an implicit `cache: true`? Maybe something for a follow-up.

```yml
- uses: denoland/setup-deno@v2
  with:
    cache-hash: ${{ hashFiles('**/deno.lock') }}
```

Reply:
In addition to that, maybe doing:

```yaml
with:
  cache: true
```

should be an implicit `cache-hash: ${{ hashFiles('**/deno.lock') }}` and someone can disable this by doing `cache-hash: ''`

Reply:
Discussed internally and we'd like to make this automatically hash the deno.lock by default similar to how https://github.com/Swatinem/rust-cache does it by default. I'll open a PR that adds this shortly after merging this one then do a release.

Reply:
Ahh, yeah that could be nice; if `cache-hash` is set, `cache: true` is implied. I can for sure do a follow-up PR with that 😃 

Note worth mentioning, `cache-hash` is not mandatory. I just had it in the README.md as I think it's best practice. So it's possible to just use `cache: true`. The primary cache key should then become `` `deno-cache-${RUNNER_OS}-${RUNNER_ARCH}-${GITHUB_JOB}-` ``

Reply:
> Discussed internally and we'd like to make this automatically hash the deno.lock by default similar to how https://github.com/Swatinem/rust-cache does it by default. I'll open a PR that adds this shortly after merging this one then do a release.

Ahh, sure. Sounds like a good idea. I just looking at `actions/setup-node`, and did not look much at others. Improving defaults and having less boilerplate sounds great!

Reply:
Added here: https://github.com/denoland/setup-deno/pull/98